baseapp/abci.go

```diff
@@ -1320,7 +1320,7 @@ func (app *BaseApp) CreateQueryContextWithCheckHeader(height int64, prove, check
 // need historical blocks.
 func (app *BaseApp) GetBlockRetentionHeight(commitHeight int64) int64 {
 	// pruning is disabled if minRetainBlocks is zero
-	if app.minRetainBlocks == 0 {
+	if app.minRetainBlocks == 0 || app.minRetainBlocks >= uint64(commitHeight) {
```

Comment:
```suggestion
	retentionBlockWindow := commitHeight - int64(app.minRetainBlocks)
	if app.minRetainBlocks == 0 || retentionBlockWindow <= 0 {
```

Then we can reuse this variable lower


```go
	retentionHeight = minNonZero(retentionHeight, retentionBlockWindow)
```

Reply:
yup applied 👍